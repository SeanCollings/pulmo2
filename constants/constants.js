export const COLORS = {
  PRIMARY: '#005bbb',
  SECONDARY: '#002f56',
  TERTIARY: '#00b8bb',
  QUARTERNARY: '#5bacf3',
  QUINERY: '#990000',
  TEXT: '#3e4b57',
  BACKGROUND: 'white',
  SECONDARY_TEXT: 'white',
  BORDER: '#CCC',
  ERROR: '#ff1507',
};
export const COLORS_DARK = {
  DARK: true,
  PRIMARY: '#222222',
  SECONDARY: '#2f9fd0',
  TERTIARY: '#28D6C0',
  QUARTERNARY: '#ff895e',
  QUINERY: '#990000',
  TEXT: '#ededed',
  BACKGROUND: '#121212',
  SECONDARY_TEXT: '#ededed',
  BORDER: '#afafaf',
  ERROR: '#fb3c36',
  DARK_GRAY: '#707070',
  DARKER_GRAY: '#2d2d2d',
};
// export const COLORS_DARK = {
//   PRIMARY: '#222222',
//   SECONDARY: '#2f9fd0',
//   TERTIARY: '#28D6C0',
//   QUARTERNARY: '#F12B7E',
//   QUINERY: '#990000',
//   TEXT: 'white',
//   BACKGROUND: '#121212',
//   SECONDARY_TEXT: 'white',
//   BORDER: 'white',
// };

// export const COLORS_DARK = {
//   PRIMARY: '#07121B',
//   SECONDARY: '#28D6C0',
//   TERTIARY: '#F12B7E',
//   TEXT: '#CCC',
//   BACKGROUND: '#07121B',
//   SECONDARY_TEXT: '#6e6e6e',
// };

export const APP_ID = '@Plumo2:';
export const TOTAL_DIFFICULTY_LEVELS = 5;
export const LIGHT_MODE = 'light_mode';
export const DARK_MODE = 'dark_mode';

export const INTIAL_STATE = 'initial';
export const BREATHING_STATE = 'breathing';
export const REST_STATE = 'rest';
export const END_STATE = 'end';

export const ROUND_STATES = [
  INTIAL_STATE,
  BREATHING_STATE,
  REST_STATE,
  END_STATE,
];

export const INSTRUCTIONS = {
  [INTIAL_STATE]: {
    state: INTIAL_STATE,
    main: '',
    sub: `rounds total`,
    prompt: 'tap to begin',
  },
  [BREATHING_STATE]: {
    state: BREATHING_STATE,
    main: '',
    sub: '',
    prompt: ' Inhale & Exhale',
  },
  [REST_STATE]: {
    state: REST_STATE,
    main: 'Rest period',
    sub: '',
    prompt: '',
  },
  [END_STATE]: {
    state: END_STATE,
    main: 'Your training is complete',
    sub: '',
    prompt: 'Complete',
  },
};
