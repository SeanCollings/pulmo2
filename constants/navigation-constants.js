import { Platform } from 'react-native';

export const defaultNavOptions = (theme) => ({
  headerStyle: {
    backgroundColor: theme.PRIMARY,
    borderBottomWidth: 1,
    borderBottomColor: theme.BORDER,
    shadowColor: 'transparent',
    shadowOffset: 0,
    shadowOpacity: 0,
  },
  headerTitleStyle: {
    fontFamily: 'tit-regular',
    backgroundColor:
      Platform.OS === 'android' ? theme.PRIMARY : theme.SECONDARY_TEXT,
  },
  headerTintColor:
    Platform.OS === 'android' ? theme.SECONDARY_TEXT : theme.PRIMARY,
});
