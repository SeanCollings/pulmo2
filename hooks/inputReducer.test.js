import {
  inputReducer,
  INPUT_TOGGLE,
  INPUT_BREATH,
  INPUT_REST,
  INPUT_END,
  INPUT_STOP,
} from './inputReducer';
import { INSTRUCTIONS } from '../constants/constants';

describe('inputReducer - unit tests', () => {
  const state = {
    isActive: false,
    currentRound: 0,
    instructions: INSTRUCTIONS.initial,
    countDownTime: 0,
  };

  test('case INPUT_TOGGLE:', () => {
    const action = { type: INPUT_TOGGLE, isActive: true };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('case INPUT_BREATH:', () => {
    const action = { type: INPUT_BREATH, currentRound: 2, countDownTime: 10 };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('case INPUT_REST:', () => {
    const action = { type: INPUT_REST };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('case INPUT_END:', () => {
    const action = { type: INPUT_END };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('case INPUT_STOP:', () => {
    const action = { type: INPUT_STOP };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });

  test('default:', () => {
    const action = { type: 'other' };
    const result = inputReducer(state, action);
    expect(result).toMatchSnapshot();
  });
});
