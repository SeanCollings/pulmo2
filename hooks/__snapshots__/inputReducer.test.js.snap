// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputReducer - unit tests case INPUT_BREATH: 1`] = `
Object {
  "countDownTime": 10,
  "currentRound": 2,
  "instructions": Object {
    "main": "",
    "prompt": "Inhale & Exhale",
    "state": "breathing",
    "sub": "",
  },
  "isActive": false,
}
`;

exports[`inputReducer - unit tests case INPUT_END: 1`] = `
Object {
  "countDownTime": 0,
  "currentRound": 0,
  "instructions": Object {
    "main": "Your training is complete",
    "prompt": "Complete",
    "state": "end",
    "sub": "",
  },
  "isActive": false,
}
`;

exports[`inputReducer - unit tests case INPUT_REST: 1`] = `
Object {
  "countDownTime": 0,
  "currentRound": 0,
  "instructions": Object {
    "main": "Rest period",
    "prompt": "",
    "state": "rest",
    "sub": "",
  },
  "isActive": true,
}
`;

exports[`inputReducer - unit tests case INPUT_STOP: 1`] = `
Object {
  "countDownTime": 0,
  "currentRound": 0,
  "instructions": Object {
    "main": "",
    "prompt": "tap to begin",
    "state": "initial",
    "sub": "rounds total",
  },
  "isActive": false,
}
`;

exports[`inputReducer - unit tests case INPUT_TOGGLE: 1`] = `
Object {
  "countDownTime": 0,
  "currentRound": 0,
  "instructions": Object {
    "main": "",
    "prompt": "tap to begin",
    "state": "initial",
    "sub": "rounds total",
  },
  "isActive": true,
}
`;

exports[`inputReducer - unit tests default: 1`] = `
Object {
  "countDownTime": 0,
  "currentRound": 0,
  "instructions": Object {
    "main": "",
    "prompt": "tap to begin",
    "state": "initial",
    "sub": "rounds total",
  },
  "isActive": false,
}
`;
